packages/permissions/src/components/authorized/authorized.tsx

```diff
@@ -65,21 +76,34 @@ const Authorized = (props: Props) => {
     props.actualActionRights
   );
 
-  return (
-    <React.Fragment>
-      {props.render(hasDemandedPermissions && hasDemandedActionRights)}
-    </React.Fragment>
-  );
+  const isAuthorized = props.demandedDataFence
+    ? createHasAppliedDataFence({
+        demandedDataFence: props.demandedDataFence,
+        actualPermissions: props.actualPermissions,
+        actualDataFences: props.actualDataFences,
+      })
+    : hasDemandedPermissions && hasDemandedActionRights;
+  return <React.Fragment>{props.render(isAuthorized)}</React.Fragment>;
 };
 Authorized.displayName = 'Authorized';
 Authorized.defaultProps = defaultProps;
 
 type TInjectAuthorizedOptions = {
   shouldMatchSomePermissions?: boolean;
   actionRights?: TDemandedActionRight[];
+  dataFence?: TDemandedDataFence;
 };
+
+type TResourceToApplyDataFence = {
+  id: string;
+};
+
 const injectAuthorized = <
-  OwnProps extends { isAuthorized?: boolean },
+  OwnProps extends {
+    isAuthorized?:
+      | boolean
+      | ((resourceToApplyDataFence: TResourceToApplyDataFence) => boolean);
```

Comment:
Maybe name this `authorizationContext` and have the type be `type TAuthorizationContext = TResourceToApplyDataFence` so we can add things to the union type later?

Reply:
yes sure thing